### PR TITLE
Added occ install option for database-port

### PR DESCRIPTION
**Pull request by Faldon**
# Implemented non-standard database port on occ install
- Added optional input option --database-port in maintenance:install command
- Refactored the databases to store the database port as string, defaulting to an empty string
- Kept parsing hostname for port option in psql and mysql when no port was provided
- Changed the PostgreSQL connection error message
### Tested
- with MySQL TCP port
- with MySQL Unix Socket
- with PostgreSQL
### ToDo
- Testing with Oracle

Solves #271

### CODE REVIEW

File: core/Command/Maintenance/Install.php [hunk 1]
```diff
@@ -50,6 +50,7 @@ protected function configure() {
 			->addOption('database', null, InputOption::VALUE_REQUIRED, 'Supported database type', 'sqlite')
 			->addOption('database-name', null, InputOption::VALUE_REQUIRED, 'Name of the database')
 			->addOption('database-host', null, InputOption::VALUE_REQUIRED, 'Hostname of the database', 'localhost')
+			->addOption('database-port', null, InputOption::VALUE_OPTIONAL, 'Port the database is listening on')
```
Comment (nickvergessen) [inline core/Command/Maintenance/Install.php]: Should be `InputOption::VALUE_REQUIRED`

Options are all optional, but when you specify it, it needs a value.

File: lib/private/Setup/OCI.php [hunk 2]
```diff
@@ -63,12 +63,14 @@ public function validate($config) {
 
 	public function setupDatabase($username) {
 		$e_host = addslashes($this->dbHost);
+		// adding slashes for security reasons
+		$e_port = addslashes($this->dbPort);
```
Comment (nickvergessen) [inline lib/private/Setup/OCI.php]: cast to int instead?

File: lib/private/Setup/PostgreSQL.php [hunk 3]
```diff
@@ -34,8 +34,11 @@ public function setupDatabase($username) {
 		$e_user = addslashes($this->dbUser);
 		$e_password = addslashes($this->dbPassword);
 
-		// Fix database with port connection
-		if(strpos($e_host, ':')) {
+		// adding port support through installer
+		if(!empty($this->dbPort)) {
+			// adding slashes for security reasons
+			$port = addslashes($this->dbPort);
```
Comment (nickvergessen) [inline lib/private/Setup/PostgreSQL.php]: same here

**Comment by Faldon** [comment]
Ofc, casting to int is much more appropriate. And in case the port was provided with the hostname, addslashes is already executed on the string.

**Comment by nickvergessen** [comment]
:+1:

**Comment by MorrisJobke** [comment]
Tested and works 👍

**Pull request merged**